src/Doctrine/Orm/Filter/AttributeFilter.php

```diff
@@ -204,15 +200,12 @@ private function getAttributeId(array $values, string $property): ?string
             return null;
         }
 
-        return $attributeId;
+        return (string) $values[self::ATTRIBUTE_ID];
     }
 
     private function extractValue(array $values, string $property): ?string
     {
         if (array_key_exists(self::VALUE, $values)) {
```

Comment:
Shouldn't this if statement be inversed now?

Reply:
I've added `!` to the if statement